src/varname.jl

```diff
@@ -294,7 +377,7 @@ function vsym end
 
 vsym(expr::Symbol) = expr
 function vsym(expr::Expr)
-    if Meta.isexpr(expr, :ref)
+    if Meta.isexpr(expr, :ref) || Meta.isexpr(expr, :.)
```

Comment:
We might also need to handle `GlobalRef` here.

Reply:
Maybe the more stable option is to use `parse_obj_lens` first to split the `vsym` and the indexing, and then just operate on the constructed VarName?  Now that the `vinds` is finally going to be removed from the tilde functions, is there any more need for `vsym`/`vinds` on their own?

Reply:
I agree with the usage of `parse_obj_lens`; that will simplify things:)

Regarding `vsym`/`vinds`, I don't know :confused: IMO `vsym` can be nice still, as it's just a nice way of extracting the symbol used as a reference, but `vinds` I agree is useless now.